### PR TITLE
feat: pause the roll automatically if it's older then 4 days

### PR DESCRIPTION
**Pull request by codebytere**
From [Slack conversation](https://electronhq.slack.com/archives/C5VT8SQ8K/p1618463441075400)

> By autopausing the roller on Thursday if the PR has been open since Monday we can guaruntee the last 2 days don't have new changes and should entirely be used as clean-up / catch-up and focusing on getting that build green so we can merge.  It also leans heavily into the desired cadence of one-pr-per-person and one-pr-per-week.

This adjusts roller to that goal.

### CODE REVIEW

**Review by VerteDinde — APPROVED** [review]

File: src/constants.ts [hunk 1]
```diff
@@ -11,15 +11,19 @@ export const REPOS = {
 
 export const NUM_SUPPORTED_VERSIONS = 4;
 
+export const FOUR_DAYS_AGO = 4 * 24 * 60 * 60 * 1000;
```
Comment (MarshallOfSound) [inline src/constants.ts:14]: Can we make this `3.9` days just to handle the couple of minutes of lag / drift that could occur around the cron job.

**Review by MarshallOfSound — CHANGES_REQUESTED** [review]
I really want to codify in the "Thursday" aspect as well as the 4 days part.  I think the rule I'm aiming for here is we should pause if both conditions are true.

* The roller PR has been open for at least 4 days
* The roller PR has been open during at least one Thursday including today

The use case for the thursday requirement is let's say hypothetically we had a really good roll and landed it on Thursday in Week N.  On Friday a new PR will be opened, the next week, this current logic will pause the roll on Monday morning (which basically makes the roll empty).

I want to try guide the pause state towards the "land one PR a week" goal and pausing on thursday gives us that "3 days of roll" + "2 days of clean up and solving the hard problems" that I'm aiming for.

**Comment by codebytere** [comment]
@MarshallOfSound fixed up - also updated Jest & other deps so i could mock the date.

**Review by nornagon — CHANGES_REQUESTED** [review]
We used to have the roller auto-pause and we decided to remove that behavior in favor of manual pausing. The reason we removed it is that roller PRs ended up getting severely out of date. I suspect we will see the same behavior again if we land this, so I'm 👎 .

**Pull request closed without merging**